### PR TITLE
Charmbrowser shapes

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
[Charmbrowser shapes](https://trello.com/c/ztke8nr9/719-charmbrowser-shapes)

**Comment by huwshimi** [comment]
QA:
- open the store
- search for something
- view the details page for a charm
- view the details page for a bundle
- there should be no console errors for any of the above

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8759/

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js [hunk 1]
```diff
@@ -160,20 +159,23 @@ class Charmbrowser extends React.Component {
...
         // TODO frankban: do we still really need this?
         const id = currentState.store || `~${currentState.user}`;
+        const { charmstore } = this.props;
         activeChild = (
           <EntityDetails
             acl={this.props.acl}
             addNotification={this.props.addNotification}
-            apiUrl={this.props.apiUrl}
             changeState={changeState}
+            charmstore={shapeup.addReshape({
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: `charmstore={this.props.charmstore.reshape(EntityDetails.propTypes.api)}` ?

**Comment by frankban** [comment]
QA ok

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: Thanks, good suggestion.

**Comment by huwshimi** [comment]
Thanks @frankban :shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8761/

**Comment by huwshimi** [comment]
Thanks @frankban :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**